File: day18/DayEighteen.java
public class DayEighteen
{
	class Node
	{
		int data;
		Node next;
		private Node(int data)
		{
			this.data = data;
			this.next = null;
		}
	}

	private Node head = null;
	private Node last = null;
	private int itr = 0;
	private int arr[] = {10,78,34,12,54,88,55};

	private void createList()
	{
		for(int itr=0;itr<arr.length;itr++)
		{
			Node newnode = new Node(arr[itr]);
			if(head == null)
			{
				head = newnode;
				last = head;
			}
			else
			{
				last.next = newnode;
				last = last.next;
			}
		}
	}

	private void createLoop()
	{
		Node last = head;
		while(last.next != null)
		{
			last = last.next;
		}
		last.next = head.next.next;
	}

	private void removeLoop()
	{
		Node slow = head;
		Node fast = head;
		while(1 == 1)
		{
			slow = slow.next;
			fast = fast.next.next;
			if(slow == fast)
			{
				break;
			}
		}

		fast = head;

		while(slow.next != fast.next)
		{
			slow = slow.next;
			fast = fast.next;
		}

		slow.next = null;
	}

	private void printList()
	{
		Node temp = head;
		while(temp != null)
		{
			System.out.print(temp.data + " ");
			temp = temp.next;
		}
		System.out.println();
	}

	public static void main(String args[])
	{
		DayEighteen list = new DayEighteen();
		list.createList();
		list.printList();
		System.out.println("--------------");
		list.createLoop();
		list.removeLoop();
		list.printList();
		System.out.println("--------------");
	}
}

File: day18/day_eighteen.cpp
# include <iostream>

using namespace std;

typedef struct Node
{
	int data;
	struct Node* next;
}node;

int itr = 0;
int arr[] = {10,78,34,12,54,88,55};
int size = sizeof(arr)/sizeof(arr[0]);

node* getNode()
{
	node* newnode = (node*)malloc(sizeof(node));
	return newnode;
}

void readNode(node* newnode)
{
	newnode->data = arr[itr];
	newnode->next = NULL;
	itr = itr + 1;
}

void createList(node** headptr)
{
	node* head = *headptr;
	node* last = *headptr;
	for(int i=0;i<size;i++)
	{
		node* newnode = getNode();
		readNode(newnode);
		if(head == NULL)
		{
			head = newnode;
			last = head;
		}
		else
		{
			last->next = newnode;
			last = last->next;
		}
	}
	*headptr = head;
}

void createLoop(node** headptr)
{
	node* last = *headptr;
	while(last->next != NULL)
	{
		last = last->next;
	}

	last->next = (*headptr)->next->next;
}

void removeLoop(node** headptr)
{
	node* slow = *headptr;
	node* fast = *headptr;
	while(1 == 1)
	{
		slow = slow->next;
		fast = fast->next->next;
		if(slow == fast)
		{
			break;
		}
	}
	fast = *headptr;
	while(slow->next != fast->next)
	{
		slow = slow->next;
		fast = fast->next;
	}
	slow->next = NULL;
}

void printList(node* temp)
{
	while(temp != NULL)
	{
		cout << temp->data << " ";
		temp = temp->next;
	}
	cout << endl;
}

int main()
{
	node* headptr = NULL;
	createList(&headptr);
	printList(headptr);
	cout << "--------------" << endl;
	createLoop(&headptr);
	removeLoop(&headptr);
	printList(headptr);
	cout << "--------------" << endl;

	return 0;
}